sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -66,6 +66,19 @@ case class CreateDatabaseCommand(
   extends RunnableCommand {
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
+    // SPARK-25464 fail if DB location exists and is not empty
+    if (path.isDefined) {
```

Comment:
not sure if we should do it here or in the external catalog, cc @gatorsmile

Reply:
The reason I did here is, external catalog always have path (default or user specified),so we end up in checking the path always.This may be the costly operation if the file system is S3

Reply:
Just want to confirm it what is the behavior of Hive (3.x) when we try to create a database in a non-empty file path?

Reply:
Please try the following cases:
- the directory does not exist and the parent directory does not exist too.  
- the directory exists but empty
- the directory exists and non-empty
- the directory is not accessible

Reply:
I think we should only block the last two cases. I also prefer to doing it in the external catalog. In the future, if we support catalog federation, different external catalog can define their own behavior. 

Listing files in S3 is expensive, but create a new database is not frequent. I think our users can accept this cost.

Reply:
> Just want to confirm it what is the behavior of Hive (3.x) when we try to create a database in a non-empty file path?

User can create the Database on the non-empty file path,results are below

host1:/opt/tools/hadoop3.1/hadoop-3.1.1/bin # ./hadoop fs -ls /user/hive/
18/09/27 18:05:39 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Found 2 items
drwxr-xr-x   - root supergroup          0 2018-09-27 18:05 /user/hive/dropTest
drwxr-xr-x   - root supergroup          0 2018-09-27 17:19 /user/hive/warehouse

hive> create database testjira location '/user/hive/';
OK
Time taken: 0.187 seconds
hive> desc database testjira;
OK
testjira                hdfs://localhost:9000/user/hive root    USER
Time taken: 0.045 seconds, Fetched: 1 row(s)
hive> drop database testjira;
OK
Time taken: 0.194 seconds
hive> 

host1:/opt/tools/hadoop3.1/hadoop-3.1.1/bin # ./hadoop fs -ls /user
18/09/27 18:07:32 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
host1:/opt/tools/hadoop3.1/hadoop-3.1.1/bin #

Reply:
> Please try the following cases:
> 
>   * the directory does not exist and the parent directory does not exist too.
spark-sql> create database noparent location '/user/dropTest/noparent/dblocation';
Time taken: 2.246 seconds
> * the directory exists but empty

spark-sql> create database emptydir location '/user/dropTest/emptydir';
Time taken: 0.262 seconds

> * the directory exists and non-empty
spark-sql> create database dirExistsAndNotempty location '/user/dropTest/';
Error in query: Cannot create Database to the location which is not empty.;

> * the directory is not accessible

spark-sql> create database userNoPermission location '/user/hive/dropTest';
Error in query: org.apache.hadoop.hive.ql.metadata.HiveException: MetaException(message:java.security.AccessControlException: Permission denied: user=sandeep, access=WRITE, inode="/user/hive":hive:hive:drwxr-x---
        at org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.check(FSPermissionChecker.java:399)
        at org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.checkPermission(FSPermissionChecker.java:261)